Frontend/ui-customer/src/Flow.purs

```diff
@@ -177,11 +179,38 @@ baseAppFlow (GlobalPayload gPayload) refreshFlow = do
               case triggerSignatureOtpResp.token of
                 Just token -> setValueToLocalStore REGISTERATION_TOKEN token
                 Nothing -> pure unit
-              currentFlowStatus
-            Left err -> do
-              liftFlowBT $ pure $ runFn3 emitJOSEvent "java" "onEvent" $ encode $  EventPayload { event : "signature_auth_failed", payload : Nothing}
-              pure unit
-        Nothing -> if (showCarouselScreen FunctionCall) then welcomeScreenFlow else enterMobileNumberScreenFlow
+              handleDeepLinks (GlobalPayload merchantPayload) false
+        Left err -> do
+          liftFlowBT $ pure $ runFn3 emitJOSEvent "java" "onEvent" $ encode $  EventPayload { event : "signature_auth_failed", payload : Nothing}
+          pure unit
+    Nothing -> if getValueToLocalStore REGISTERATION_TOKEN /= "__failed" && getValueToLocalStore REGISTERATION_TOKEN /= "(null)" &&  (isNothing $ payload.signatureAuthData)
+    then handleDeepLinks (GlobalPayload merchantPayload) false
+    else if (showCarouselScreen FunctionCall) then welcomeScreenFlow else enterMobileNumberScreenFlow
+
+handleDeepLinks :: GlobalPayload -> Boolean -> FlowBT String Unit
+handleDeepLinks (GlobalPayload merchantPayload) shouldSkipDefault = do
+  let (Payload payload) = merchantPayload.payload
+  case payload.view_param of
+    Just screen -> case screen of
+      "rides" -> hideSpalshAndCallFlow $ myRidesScreenFlow true
```

Comment:
can we have short forms of screens instead of having more characters? good to have smaller links

Reply:
we will shorten the longer ones.

Reply:
probably have a coding all, instead of just for the longer ones